src/ProjectFileManager.cpp

```diff
@@ -1306,6 +1308,26 @@ class ImportProgress final
 
    }
 
+   void OnImportStarted(const wxString &filename) override
+   {
+       mImportInfoDialog = new ImportInfoDialog(filename, NULL, -1, XO("Importing files"));
+       mImportInfoDialog->Show();
+       wxYield();
+   }
+    
+   bool OnCheckImportCancelled() override
+   {
+        return mImportInfoDialog->IsCancelled();
+   }
+    
+   void OnImportStartCleanup() override
```

Comment:
Doesn't need to be a part of interface. Please consider using `ProjectWindows` or assigning this dialog directly to a project frame via `GetProjectFrame`

Reply:
assigned dialog to project frame but left function and renamed it (still need to call EndModal)